### PR TITLE
test: proposed tasks tests

### PR DESCRIPTION
**Pull request by karel-rehor**
Preliminary unit and integration tests on tasks.  

To open discussion on how Bonitoo should be writing and contributing tests.

### CODE REVIEW

File: src/tasks/reducers/tasks.test.ts [hunk 1]
```diff
@@ -53,4 +53,36 @@ describe('tasksReducer', () => {
       expect(actual).toEqual(expected)
     })
   })
+
+  // N.B. Started as learning exercise.  low value?
+  // Seems ADD_TASK simply wraps resource/reducers/helpers
+  // Which are already covered
```
Comment (genehynson) [inline src/tasks/reducers/tasks.test.ts:59]: I agree with this comment, this test probably isn't necessary

**Comment by genehynson** [comment]
@karel-rehor can you try rebasing this fork with the UI repo's master branch? The CI should run if you do that. 👍

**Review by Asalem1 — COMMENTED** [review]
I'm a little curious where the ask for these tests came from? I understand and value the level of effort to try and add some of these unit tests onto the platform, but I feel like we have a lot of this code covered in some of our e2e tests and that this might be a bit redundant. What's the point of pursuing unit level stability in addition to our cypress tests?

**Comment by genehynson** [comment]
@karel-rehor thanks for rebasing the branch! I'll look into why the pipeline isn't running for you tomorrow

**Comment by genehynson** [comment]
@karel-rehor monitor-ci will stick around but will not be used for local development any more. We are using it to run our CI tests. You probably won't be able to run it locally either. The issue is that the pipeline doesn't know how to clone the forked UI repo so I'm making some adjustments. We need the monitor-ci-tests to pass before any PR can be merged in.

**Comment by genehynson** [comment]
Ok I have the necessary changes to fix the pipeline in a PR and will get that reviewed & merged tomorrow.

**Comment by genehynson** [comment]
Ok @karel-rehor sorry about the delay - CircleCI had an outage on Friday. If you rebase this branch with the `master` branch from origin, the CI should work now.

**Comment by karel-rehor** [comment]
@genehynson Hi Gene, I rebased our fork a couple of days ago, but I see the monitor-ci-tests still fail. 

```
#!/bin/bash -eo pipefail
API_KEY=${MONITOR_CI_API_KEY} PULL_REQUEST=${CIRCLE_PULL_REQUEST} MONITOR_CI_BRANCH="master" UI_BRANCH=${CIRCLE_BRANCH} SHA=${CIRCLE_SHA1} ./scripts/run-monitor-ci-tests.bash

no passing monitor-ci pipelines found for this SHA, starting a new one

starting monitor-ci pipeline targeting monitor-ci branch master, UI branch jest/experim/kk and using UI SHA 677f8721d622f5624b3662f7cb9283b05ae5cef8

Exited with code exit status 22
CircleCI received exit code 22
```

Is there anything we can do on our side to get this check to pass?

**Comment by genehynson** [comment]
> @genehynson Hi Gene, I rebased our fork a couple of days ago, but I see the monitor-ci-tests still fail.
> 
> ```
> #!/bin/bash -eo pipefail
> API_KEY=${MONITOR_CI_API_KEY} PULL_REQUEST=${CIRCLE_PULL_REQUEST} MONITOR_CI_BRANCH="master" UI_BRANCH=${CIRCLE_BRANCH} SHA=${CIRCLE_SHA1} ./scripts/run-monitor-ci-tests.bash
> 
> no passing monitor-ci pipelines found for this SHA, starting a new one
> 
> starting monitor-ci pipeline targeting monitor-ci branch master, UI branch jest/experim/kk and using UI SHA 677f8721d622f5624b3662f7cb9283b05ae5cef8
> 
> Exited with code exit status 22
> CircleCI received exit code 22
> ```
> 
> Is there anything we can do on our side to get this check to pass?

Hi @karel-rehor we had to revert my changes to the pipeline because they broke other things. I'm working on a PR today to resolve it and will get it merged in again

**Comment by genehynson** [comment]
@karel-rehor I also noticed bonitoo has setup a circleCI project for this fork of the UI repo - that won't work! We need to run these PRs through our [Influxdata circle instance](https://app.circleci.com/pipelines/github/influxdata/ui). This will happen by default if you remove this project from the bonitoo circle instance.

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 45 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=1411)

**Comment by genehynson** [comment]
Yay! The CI works! 🎉

**Review by genehynson — APPROVED** [review]

**Pull request merged**